FRAuth/FRAuth/DeviceBinding/SharedPreferencesDeviceRepository.swift

```diff
@@ -0,0 +1,88 @@
+// 
+//  SharedPreferencesDeviceRepository.swift
+//  FRAuth
+//
+//  Copyright (c) 2022 ForgeRock. All rights reserved.
+//
+//  This software may be modified and distributed under the terms
+//  of the MIT license. See the LICENSE file for details.
+//
+
+
+import Foundation
+import FRCore
+
+
+protocol DeviceRepository {
+    
+    /// Persist the data in encrypted shared preference
+    func persist(userId: String,
+                 userName: String,
+                 key: String,
+                 authenticationType: DeviceBindingAuthenticationType) throws -> String
+    
+    func getAllKeys() -> [String: Any]?
+}
+
+
+/// Helper class to save and retrieve EncryptedMessage
+internal class SharedPreferencesDeviceRepository: DeviceRepository {
```

Comment:
I think we better rename this class, understand that we are trying to align with Android, "SharedPreferences" is a class for Android to store key/value. If we store in KeyChain, we can call it KeyChainDeviceRespsitory.

Reply:
+1

Reply:
Renamed to **KeychainDeviceRepository**